modules/acquia_cms_search/src/Facade/SearchFacade.php

```diff
@@ -290,4 +316,44 @@ private function loadIndexFromSettings(ThirdPartySettingsInterface $object) : ?I
     return $index ? $this->indexStorage->load($index) : NULL;
   }
 
+  /**
+   * Check if reference field is of allowed type.
+   *
+   * @param \Drupal\field\FieldStorageConfigInterface $field_storage
+   *   The field configuration object.
+   *
+   * @return bool
+   *   The boolean value based on condition.
+   */
+  public function isAllowedTypeReferenceField(FieldStorageConfigInterface $field_storage): bool {
+    return ($field_storage->getType() === 'entity_reference' && in_array($field_storage->getSetting('target_type'), self::TARGET_REFERENCE_TYPES));
+  }
+
+  /**
+   * Remove field from index.
+   *
+   * @param \Drupal\field\FieldStorageConfigInterface $field_storage
+   *   The field configuration object.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   * @throws \Drupal\search_api\SearchApiException
+   * @throws \Exception
+   */
+  public function removeFieldFromIndex(FieldStorageConfigInterface $field_storage): void {
+    /** @var \Drupal\search_api\IndexInterface $index */
+    $index = $this->loadIndexFromSettings($field_storage);
+    $filed_name = $field_storage->getName();
+    $field = $index->getField($filed_name);
+    if (!$field) {
```

Comment:
we can replace if (!$index->getField($filedName)) { and remove line# 346 as it is not using further.